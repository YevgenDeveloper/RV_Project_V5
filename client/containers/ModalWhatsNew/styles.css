@import 'client/styles/variables';

.button {
  display: block;
  width: 240px;
  height: 40px;
  margin: 30px auto 0;
}
