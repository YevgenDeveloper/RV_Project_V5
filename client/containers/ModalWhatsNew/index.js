// @flow
import React from 'react';

import { Modal, Button } from 'client/components';
import styles from './styles.css';

type Props = {
  whatsnew?: string[],
  close: void => void,
};

const ModalWhatsNew = (props: Props) => (
  <Modal close={props.close}>
    <h1>Whatâ€™s new</h1>
    {props.whatsnew &&
      props.whatsnew.map((string, index) => (
        /* eslint-disable react/no-array-index-key */
        <p key={index}>{string}</p>
      ))}
    <Button
      blue
      label="Continue"
      onClick={() => props.close()}
      className={styles.button}
    />
  </Modal>
);

ModalWhatsNew.defaultProps = {
  whatsnew: [],
};

export default ModalWhatsNew;
