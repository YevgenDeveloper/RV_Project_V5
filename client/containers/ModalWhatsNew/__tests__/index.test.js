// @flow

import type { ShallowWrapper } from 'enzyme';

import React from 'react';
import { shallow } from 'enzyme';
import ModalWhatsNew from '..';

it('renders correctly', () => {
  const wrapper: ShallowWrapper = shallow(<ModalWhatsNew close={jest.fn()} />);

  expect(wrapper).toMatchSnapshot();
});
