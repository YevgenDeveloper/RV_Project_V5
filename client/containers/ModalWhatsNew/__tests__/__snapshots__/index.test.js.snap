// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalWhatsNew
    close={[MockFunction]}
    whatsnew={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <h1>
          What’s new
        </h1>,
        Array [],
        <Button
          blue={true}
          className="button"
          dark={false}
          disabled={false}
          label="Continue"
          onClick={[Function]}
        />,
      ],
      "className": undefined,
      "close": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "What’s new",
        },
        "ref": null,
        "rendered": "What’s new",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "blue": true,
          "className": "button",
          "dark": false,
          "disabled": false,
          "label": "Continue",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <h1>
            What’s new
          </h1>,
          Array [],
          <Button
            blue={true}
            className="button"
            dark={false}
            disabled={false}
            label="Continue"
            onClick={[Function]}
          />,
        ],
        "className": undefined,
        "close": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "What’s new",
          },
          "ref": null,
          "rendered": "What’s new",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "blue": true,
            "className": "button",
            "dark": false,
            "disabled": false,
            "label": "Continue",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
